app/Domain/Clients/Controllers/NewClient.php

```diff
@@ -10,35 +10,61 @@
     use Leantime\Core\Controller\Frontcontroller;
     use Leantime\Domain\Auth\Models\Roles;
     use Leantime\Domain\Auth\Services\Auth;
-    use Leantime\Domain\Clients\Repositories\Clients as ClientRepository;
-    use Leantime\Domain\Users\Repositories\Users as UserRepository;
+    use Leantime\Domain\Clients\Services\Clients as ClientService;
+    use Symfony\Component\HttpFoundation\Response;
 
     class NewClient extends Controller
     {
-        private ClientRepository $clientRepo;
-
-        private UserRepository $user;
+        private ClientService $clientService;
 
         /**
          * init - initialize private variables
          */
-        public function init(ClientRepository $clientRepo, UserRepository $user)
+        public function init(
+            ClientService $clientService 
+        ){
+            $this->clientService = $clientService;
+        }
+
+        /**
+         * get - display template and provide empty values object
+         */
+        public function get(): Response
         {
+            Auth::authOrRedirect([Roles::$owner, Roles::$admin], true);
+
+            //Only admins
+            if (Auth::userIsAtLeast(Roles::$admin)) {
+                $values = (object) [
```

Comment:
Instead of creating the array/object manually you should use the newly create client model to prepare an empty object. `$client = app()->make(ClientModel::class);`

On reason to use models is so that we don't have to worry about array definitions. For example, if I want to add a new field I shouldn't have to look for arrays in controllers. I just add it to the model and it will be available everywhere